src/EventListener/DeserializeListener.php

```diff
@@ -66,12 +66,31 @@ public function onKernelRequest(GetResponseEvent $event)
         $request->attributes->set(
             'data',
             $this->serializer->deserialize(
-                $request->getContent(), $attributes['resource_class'], $format, $context
+                $this->getRequestContent($request), $attributes['resource_class'], $format, $context
```

Comment:
If we skip this step, for example with
"        $request->attributes->set(
            'data',
            new \stdClass() (or anything you want)
We lost the loaded entity (we can't retrieve it in "data") in the case of empty put request on specific entity (for exemple PUT /dummies/1/custom_operation)

And if you just skip the $request->attributes->set, the encoding exception is thrown.